backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardlist/stack/internal/BpkStackCardListImpl.kt

```diff
@@ -0,0 +1,136 @@
+    accessoryStyle: BpkStackCardAccessoryStyle? = null,
+    headerButton: BpkSectionHeaderButton? = null,
+    accessibilityHeaderTagEnabled: Boolean? = true,
+    content: @Composable ((Int) -> Unit),
+) {
+
+    var isExpanded by rememberSaveable { mutableStateOf(false) }
+    val targetCount = when (accessoryStyle) {
+        is BpkStackCardAccessoryStyle.Expand -> {
+            if (accessoryStyle.expandedCount < 0 || accessoryStyle.collapsedCount < 0) {
+                throw IllegalArgumentException("expandedCount/collapsedCount can not be negative")
+            }
+            if (accessoryStyle.expandedCount < accessoryStyle.collapsedCount) {
+                throw IllegalArgumentException("expandedCount should be greater than collapsedCount")
+            }
+            if (accessoryStyle.expandedCount <= MINIMUM_EXPANDED_COUNT) {
+                totalCount
+            } else {
+                if (isExpanded) accessoryStyle.expandedCount else accessoryStyle.collapsedCount
+            }
+        }
+        else -> totalCount
+    }
+
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = PaddingValues(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+    ) {
+        item {
+            BpkSectionHeader(
+                title = title,
+                description = description,
+                modifier = Modifier,
+                button = headerButton,
+                accessibilityHeaderTagEnabled = accessibilityHeaderTagEnabled,
+            )
+        }
+
+        items(targetCount) {
+            Box(Modifier.animateItem()) {
+                content(it)
+            }
+        }
+
+        item {
+            accessoryStyle?.let {
+                when (it) {
+                    is BpkStackCardAccessoryStyle.Expand -> {
+                        if (it.expandedCount > MINIMUM_EXPANDED_COUNT) {
+                            BpkButton(
+                                modifier = Modifier
+                                    .fillMaxWidth()
+                                    .padding(bottom = BpkSpacing.Base),
+                                text = if (isExpanded) it.collapsedText else it.expandText,
+                                icon = if (isExpanded) BpkIcon.ChevronUp else BpkIcon.ChevronDown,
+                                position = BpkButtonIconPosition.End,
+                                type = BpkButtonType.Link,
+                                onClick = {
+                                    isExpanded = !isExpanded
+                                    it.onExpansionChange?.invoke(isExpanded)
+                                },
+                            )
+                        }
+                    }
+
+                    is BpkStackCardAccessoryStyle.Button -> {
+                        if (headerButton == null) {
+                            BpkButton(
+                                modifier = Modifier
```

Comment:
We should add `animateItem()` here as well, as the button will need to move down after adding a new item.
```suggestion
                                modifier = Modifier
                                    .animateItem()
```

Reply:
See [Henrik's comment above](https://github.com/Skyscanner/backpack-android/pull/2257#discussion_r2039537345) which includes this fix **_but better_**

Reply:
done